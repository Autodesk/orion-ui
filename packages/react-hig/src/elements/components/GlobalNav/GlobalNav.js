/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as HIG from 'hig.web';
import * as PropTypes from 'prop-types';

import HIGElement from '../../HIGElement';
import HIGChildValidator from '../../HIGChildValidator';
import createComponent from '../../../adapters/createComponent';

import SideNavComponent, { SideNav } from './SideNav';
import TopNavComponent, { TopNav } from './TopNav';
import SubNavComponent, { SubNav } from './SubNav';
import Slot from './Slot';

class GlobalNav extends HIGElement {
  constructor(initialProps) {
    super(HIG.GlobalNav, initialProps);
  }

  componentDidMount() {
    // Add any children
    if (this.sideNav) {
      this.hig.addSideNav(this.sideNav.hig);
      this.sideNav.componentDidMount();
    }

    if (this.topNav) {
      this.hig.addTopNav(this.topNav.hig);
      this.topNav.componentDidMount();
    }

    if (this.subNav) {
      this.hig.addSubNav(this.subNav.hig);
      this.subNav.componentDidMount();
    }

    if (this.initialProps.sideNavOpen) {
      this.hig.showSideNav();
    } else {
      this.hig.hideSideNav();
    }

    if (this.slot) {
      this.hig.addSlot(this.slot);
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case TopNav:
        return new TopNav(this.hig.partials.TopNav, props);
      case SubNav:
        return new SubNav(this.hig.partials.SubNav, props);
      case SideNav:
        return new SideNav(this.hig.partials.SideNav, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  appendChild(instance, beforeChild = {}) {
    if (instance instanceof SideNav) {
      if (this.sideNav) {
        throw new Error('only one SideNav is allowed');
      } else {
        this.sideNav = instance;
        if (this.mounted) {
          this.hig.addSideNav(instance.hig);
          instance.componentDidMount();
        }
      }
    } else if (instance instanceof TopNav) {
      if (this.topNav) {
        throw new Error('only one TopNav is allowed');
      } else {
        this.topNav = instance;
        if (this.mounted) {
          this.hig.addTopNav(instance.hig);
          instance.componentDidMount();
        }
      }
    } else if (instance instanceof SubNav) {
      if (this.SubNav) {
        throw new Error('only one SubNav is allowed');
      } else {
        this.subNav = instance;
        if (this.mounted) {
          this.hig.addSubNav(instance.hig);
          instance.componentDidMount();
        }
      }
    } else {
      throw new Error('unknown type');
    }
  }

  addSlot(element) {
    if (this.mounted) {
      this.hig.addSlot(element);
    } else {
      this.slot = element;
    }
  }

  insertBefore(instance) {
    this.appendChild(instance);
  }

  removeChild(instance) {
    if (instance instanceof SideNav) {
      this.sideNav = null;
    } else if (instance instanceof TopNav) {
      this.topNav = null;
    } else if (instance instanceof SubNav) {
      this.subNav = null;
    }
    instance.unmount();
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'sideNavOpen': {
          if (propValue) {
            this.hig.showSideNav();
          } else {
            this.hig.hideSideNav();
          }
          break;
        }
        case 'children': {
          /* no-op */
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }
}

const GlobalNavComponent = createComponent(GlobalNav);

GlobalNavComponent.propTypes = {
  sideNavOpen: PropTypes.bool,
  children: HIGChildValidator([
    SideNavComponent,
    TopNavComponent,
    SubNavComponent,
    Slot
  ])
};

GlobalNavComponent.__docgenInfo = {
  props: {
    sideNavOpen: {
      description: 'show or hide the SideNav'
    },

    children: {
      description: 'support adding SideNav, SubNav, or TopNav'
    }
  }
};

GlobalNavComponent.SideNav = SideNavComponent;
GlobalNavComponent.TopNav = TopNavComponent;
GlobalNavComponent.SubNav = SubNavComponent;
GlobalNavComponent.Slot = Slot;

export default GlobalNavComponent;
