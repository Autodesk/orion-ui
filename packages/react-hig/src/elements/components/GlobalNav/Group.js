/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import createComponent from '../../../adapters/createComponent';
import HIGElement from '../../HIGElement';
import HIGNodeList from '../../HIGNodeList';
import HIGChildValidator from '../../HIGChildValidator';
import ModuleComponent, { Module } from './Module';

export class Group extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);
    this.modules = new HIGNodeList({
      type: Module,
      HIGConstructor: this.hig.partials.Module,
      onAdd: (instance, beforeInstance) => {
        this.hig.addModule(instance, beforeInstance);
      }
    });

    this.state = {};
    this._render = this._render.bind(this);
  }

  componentDidMount() {
    this.modules.componentDidMount();
    this._render();
  }

  createElement(ElementConstructor, props) {
    return this.modules.createElement(ElementConstructor, props);
  }

  insertBefore(instance, insertBeforeIndex) {
    this.modules.insertBefore(instance, insertBeforeIndex);
    if (this.mounted) {
      this._render();
    }
  }

  removeChild(instance) {
    this.modules.removeChild(instance);
  }

  commitUpdate(updatePayload, oldProps, newProps) {
    this.processUpdateProps(updatePayload).then(this._render);
  }

  isVisible() {
    return this.state.isVisible;
  }

  expandModules() {
    this.modules.forEach(module => {
      module.commitUpdate({query: undefined})
      module.expandAll();
    });
    this._render();
  }

  collapseModules() {
    this.modules.forEach(module => {
      this.commitUpdate({query: undefined})
      module.collapseAll();
    });
    this._render();
  }

  _render() {
    const matches = this.modules.map(module => {
      module.commitUpdate({
        query: this.props.query,
        activeModule: this.props.activeModule,
        activeSubmodule: this.props.activeSubmodule
      });

      return module.isVisible();
    });

    if (matches.some(m => m)) {
      this.hig.show();
      this.state.isVisible = true;
    } else {
      this.hig.hide();
      this.state.isVisible = false;
    }
  }
}

const GroupComponent = createComponent(Group);

GroupComponent.propTypes = {
  children: HIGChildValidator([ModuleComponent])
};

GroupComponent.__docgenInfo = {
  props: {
    children: {
      description: 'support adding Module'
    }
  }
};

GroupComponent.Module = ModuleComponent;

export default GroupComponent;
