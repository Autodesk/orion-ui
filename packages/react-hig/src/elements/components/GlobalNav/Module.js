/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as PropTypes from 'prop-types';
import createComponent from '../../../adapters/createComponent';
import HIGElement from '../../HIGElement';
import HIGNodeList from '../../HIGNodeList';
import HIGChildValidator from '../../HIGChildValidator';
import SubmoduleComponent, { Submodule } from './Submodule';

export class Module extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);
    this.submodules = new HIGNodeList({
      type: Submodule,
      HIGConstructor: this.hig.partials.Submodule,
      onAdd: (instance, beforeInstance) => {
        this.hig.addSubmodule(instance, beforeInstance);
      }
    });
    this.state = {
      title: initialProps.title
    };
  }

  commitUpdate(updatePayload, oldProps, newProps) {
    const mapping = {
      icon: 'setIcon',
      title: 'setTitle',
      link: 'setLink'
    };

    this.commitUpdateWithMapping(updatePayload, mapping);

    if (Object.keys(updatePayload).includes('title')) {
      this.state.title = updatePayload.title;
    }
  }

  componentDidMount() {
    this.submodules.componentDidMount();
  }

  createElement(ElementConstructor, props) {
    return this.submodules.createElement(ElementConstructor, props);
  }

  insertBefore(instance, insertBeforeIndex) {
    this.submodules.insertBefore(instance, insertBeforeIndex);
  }

  removeChild(instance) {
    this.submodules.removeChild(instance);
  }

  matches(query) {
    return this.state.title.toLowerCase().match(query.toLowerCase());
  }

  filter(query) {
    const childMatches = this.submodules.map(submodule => {
      return submodule.filter(query);
    });

    if (childMatches.some(m => m) || this.matches(query)) {
      this.hig.show();
      return true;
    } else {
      this.hig.hide();
      return false;
    }
  }
}

const ModuleComponent = createComponent(Module);

ModuleComponent.propTypes = {
  icon: PropTypes.string,
  title: PropTypes.string,
  link: PropTypes.string,
  show: PropTypes.func,
  hide: PropTypes.func,
  activate: PropTypes.func,
  deactivate: PropTypes.func,
  showSubmodules: PropTypes.func,
  hideSubmodules: PropTypes.func,
  submodulesClosed: PropTypes.bool,
  addSubmodule: PropTypes.func,
  onClick: PropTypes.func,
  onHover: PropTypes.func,
  children: HIGChildValidator([SubmoduleComponent])
};

ModuleComponent.__docgenInfo = {
  props: {
    icon: {
      description: 'sets the icon of an Module'
    },

    title: {
      description: 'sets the title of an Module'
    },

    link: {
      description: 'sets the link of an Module'
    },

    show: {
      description: 'show (used for filtering)'
    },

    hide: {
      description: 'hide (used for filtering)'
    },

    activate: {
      description: 'activates the module'
    },

    deactivate: {
      description: 'deactivates the module'
    },

    showSubmodules: {
      description: 'show the submodules'
    },

    hideSubmodules: {
      description: 'hide the submodules'
    },

    addSubmodule: {
      description: 'add submodule'
    },

    onClick: {
      description: 'triggered when a link is clicked on'
    },

    onHover: {
      description: 'triggered when a link is hovered over'
    },

    children: {
      description: 'support adding SubModule'
    }
  }
};

ModuleComponent.Submodule = SubmoduleComponent;

export default ModuleComponent;
