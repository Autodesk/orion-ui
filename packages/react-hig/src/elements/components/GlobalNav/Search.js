/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as PropTypes from 'prop-types';
import createComponent from '../../../adapters/createComponent';
import HIGElement from '../../HIGElement';

export class Search extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    ['showClearIcon', 'hideClearIcon'].forEach(fn => { this[fn] = this[fn].bind(this) });
  }
  commitUpdate(updatePayload, oldProps, newProps) {
    const mapping = {
      placeholder: 'setPlaceholder',
      query: 'setQuery'
    };

    this.commitUpdateWithMapping(updatePayload, mapping);
  }

  componentDidMount() {
    this.hig.onFocusIn(this.showClearIcon);
    this.hig.onFocusOut(this.hideClearIcon);
  }

  showClearIcon() {
    this.hig.showClearIcon();
  }

  hideClearIcon() {
    this.hig.hideClearIcon();
  }
}

const SearchComponent = createComponent(Search);

SearchComponent.propTypes = {
  query: PropTypes.string,
  placeholder: PropTypes.string,
  showClearIcon: PropTypes.func,
  hideClearIcon: PropTypes.func,
  onClearIconClick: PropTypes.func,
  onInput: PropTypes.func,
  onFocusIn: PropTypes.func,
  onFocusOut: PropTypes.func
};

SearchComponent.__docgenInfo = {
  props: {
    placeholder: {
      description: 'sets the placeholder'
    },

    query: {
      description: 'sets the search query'
    },

    showClearIcon: {
      description: '{func} show the clear input icon'
    },

    hideClearIcon: {
      description: '{func} hide the clear input icon'
    },

    onClearIconClick: {
      description: '{func} triggered when clear icon is clicked'
    },

    onInput: {
      description: 'triggered when search field has something entered'
    },

    onFocusIn: {
      description: 'triggered when search field is in focus'
    },

    onFocusOut: {
      description: 'triggered when search field is not in focus'
    }
  }
};

export default SearchComponent;
