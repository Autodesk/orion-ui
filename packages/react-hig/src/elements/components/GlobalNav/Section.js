/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as PropTypes from 'prop-types';

import createComponent from '../../../adapters/createComponent';
import HIGElement from '../../HIGElement';
import HIGNodeList from '../../HIGNodeList';
import HIGChildValidator from '../../HIGChildValidator';
import GroupComponent, { Group } from './Group';
import CollapseComponent, { Collapse } from './Collapse';

export class Section extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);
    this.groups = new HIGNodeList({
      type: Group,
      HIGConstructor: this.hig.partials.Group,
      onAdd: (instance, beforeInstance) => {
        this.hig.addGroup(instance, beforeInstance);
      }
    });
    this.state = {};
  }

  componentDidMount() {
    this.groups.componentDidMount();
    if (this.collapse) {
      this.hig.addCollapse(this.collapse.hig);
      this.collapse.mount();
    }
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    const mapping = {
      headerLabel: 'setHeaderLabel',
      headerName: 'setHeaderName'
    };

    this.commitUpdateWithMapping(updatePayload, mapping);

    if (Object.keys(updatePayload).includes('query')) {
      this.state.query = updatePayload.query;
      this._render();
    }

    if (Object.keys(updatePayload).includes('expand')) {
      this.state.expand = updatePayload.expand;
      this._render();
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case Group:
        return this.groups.createElement(ElementConstructor, props);
      case Collapse:
        return new Collapse(this.hig.partials.Collapse, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  insertBefore(instance, insertBeforeIndex) {
    this.appendChild(instance);
  }

  appendChild(instance) {
    if (instance instanceof Group) {
      this.groups.insertBefore(instance); // calls internal _appendChild if no "before" component
    } else if (instance instanceof Collapse) {
      if (this.collapse) {
        throw new Error('only one Collapse is allowed');
      } else {
        this.collapse = instance;
        if (this.mounted) {
          this.hig.addCollapse(instance.hig);
          instance.mount();
        }
      }
    }
  }

  removeChild(instance) {
    this.groups.removeChild(instance);
  }

  _render() {
    const childVisibility = this.groups.map(group => {
      group.commitUpdate({
        query: this.state.query,
        expanded: this.state.expanded
      });

      return group.isVisible();
    });

    if (childVisibility.some(v => v)) {
      this.hig.show();
    } else {
      this.hig.hide();
    }
  }
}

const SectionComponent = createComponent(Section);

SectionComponent.propTypes = {
  headerLabel: PropTypes.string,
  headerName: PropTypes.string,
  children: HIGChildValidator([GroupComponent, CollapseComponent])
};

SectionComponent.__docgenInfo = {
  props: {
    headerLabel: {
      description: 'sets the label'
    },

    headerName: {
      description: 'sets the name'
    },

    children: {
      description: 'support adding Group and Collapse'
    }
  }
};

SectionComponent.Group = GroupComponent;
SectionComponent.Collapse = CollapseComponent;

export default SectionComponent;
