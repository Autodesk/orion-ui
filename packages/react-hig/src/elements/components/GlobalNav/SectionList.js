/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import createComponent from '../../../adapters/createComponent';
import HIGNodeList from '../../HIGNodeList';
import HIGChildValidator from '../../HIGChildValidator';

import SectionComponent, { Section } from './Section';

// Does not extend HIGElement because it's not a real HIG component
export class SectionList {
  constructor(higInstance) {
    this.hig = higInstance;
    this.sections = new HIGNodeList({
      type: Section,
      HIGConstructor: this.hig.partials.Section,
      onAdd: (instance, beforeInstance) => {
        this.hig.addSection(instance, beforeInstance);
      }
    });
  }

  mount() {
    this.sections.componentDidMount();
  }

  unmount() {
    // no-op
  }

  commitUpdate(updatePayload, oldProps, newProps) {
    // no-op
  }

  createElement(ElementConstructor, props) {
    return this.sections.createElement(ElementConstructor, props);
  }

  insertBefore(instance, insertBeforeIndex) {
    this.sections.insertBefore(instance, insertBeforeIndex);
  }

  removeChild(instance) {
    this.sections.removeChild(instance);
  }
}

const SectionListComponent = createComponent(SectionList);

SectionListComponent.propTypes = {
  children: HIGChildValidator([SectionComponent])
};

SectionListComponent.__docgenInfo = {
  props: {
    children: {
      description: 'support adding Section'
    }
  }
};

SectionListComponent.Item = SectionComponent;

export default SectionListComponent;
