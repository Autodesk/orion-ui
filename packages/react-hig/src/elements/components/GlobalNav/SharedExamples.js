/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import { mount } from 'enzyme';
import React from 'react';

class SharedExamples {
  constructor(context, createHigContext) {
    this.Context = context;
    this.createHigContext = createHigContext;
  }

  verifyPropsSet(config) {
    const defaults = { [config.key]: config.sampleValue };
    const { higContainer, higItem } = this.createHigContext(defaults);

    const reactContainer = document.createElement('div');

    const wrapper = mount(<this.Context {...defaults} />, {
      attachTo: reactContainer
    });

    expect(reactContainer.firstChild.outerHTML).toMatch(config.sampleValue);
    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();
    expect(reactContainer.firstChild.outerHTML).toEqual(
      higContainer.firstChild.outerHTML
    );
  }

  verifyPropsUpdate(config) {
    const defaults = { [config.key]: config.sampleValue };
    const { higContainer, higItem } = this.createHigContext(defaults);

    const reactContainer = document.createElement('div');

    const wrapper = mount(<this.Context {...defaults} />, {
      attachTo: reactContainer
    });

    //Update hig.web instance
    higItem[config.mutator](config.updateValue);

    //Update React Instance
    wrapper.setProps({ [config.key]: config.updateValue });

    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

    //Check them against each other
    expect(reactContainer.firstChild.outerHTML).toEqual(
      higContainer.firstChild.outerHTML
    );
  }
}

export default SharedExamples;
