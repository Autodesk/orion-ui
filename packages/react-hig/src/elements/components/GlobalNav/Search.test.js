/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/

import { mount } from 'enzyme';
import * as HIG from 'hig.web';
import React from 'react';

import GlobalNav from './GlobalNav';
import SideNav from './SideNav';
import Search from './Search';
import Section from './Section';
import SectionList from './SectionList';
import Group from './Group';
import Module from './Module';
import Submodule from './Submodule';

import SharedExamples from './SharedExamples';

const Context = props => {
  return (
    <GlobalNav>
      <SideNav>
        <Search placeholder={props.placeholder} />
        <SectionList>
          <Section>
            <Group>
              <Module tile={props.moduleTitle} icon="assets">
                <Submodule title={props.submoduleTitle} icon="assets" />
              </Module>
            </Group>
          </Section>
        </SectionList>
      </SideNav>
    </GlobalNav>
  );
};

function setUpSearch() {
  const defaults = {
    placeholder: 'Search for something',
    submoduleTitle: 'Document New',
    moduleTitle: 'Docoument Workflow'
  };

  const reactContainer = document.createElement('div');
  const wrapper = mount(<Context {...defaults} />, {
    attachTo: reactContainer
  });
  return { wrapper, reactContainer };
}

describe('<Search>', () => {
  it('will is rendered as a child of Side Nav', () => {
    const { reactContainer } = setUpSearch();

    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();
  });
});
