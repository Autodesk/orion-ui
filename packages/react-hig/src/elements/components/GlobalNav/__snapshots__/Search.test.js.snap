// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search> will is rendered as a child of Side Nav 1`] = `
"<div class=\\"hig__global-nav\\">
    <div class=\\"hig__global-nav__sidenav\\">
    <div class=\\"hig__global-nav__sidenav__scroll hig__global-nav__sidenav__scroll--search\\">
        <div class=\\"hig__global-nav__side-nav__section\\">
    <div class=\\"hig__global-nav__side-nav__section__header\\">
        <span class=\\"hig__global-nav__side-nav__section__header__label\\">Project</span>
        <span class=\\"hig__global-nav__side-nav__section__header__name\\">X</span>
        <div class=\\"hig__global-nav__side-nav__section__collapse hig__global-nav__side-nav__section__collapse--collapsed\\"><div class=\\"hig__icon\\"><svg width=\\"11\\" height=\\"12\\" viewBox=\\"0 0 11 12\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"double-caret\\"><g stroke-width=\\"1.5\\" stroke=\\"#979797\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"><polyline points=\\"8 11 12 15 16 11\\" transform=\\"translate(-6.432 -9.533)\\"></polyline><polyline points=\\"8 11 12 15 16 11\\" transform=\\"translate(-6.432 -4.533)\\"></polyline></g></svg></div></div><!--SECTION-COLLAPSE-->
    </div>
    <div class=\\"hig__global-nav__side-nav__section__group\\">
    <div class=\\"hig__global-nav__side-nav__section__group__module\\">
    <a class=\\"hig__global-nav__side-nav__section__group__module__link\\" href=\\"#\\">
        <div class=\\"hig__global-nav__side-nav__section__group__module__link__icon\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"assets\\"><title>hig-light/globalnav/sidenav/icon/assets</title><g fill=\\"#221F20\\" fill-rule=\\"evenodd\\"><path d=\\"M10.1898101,0.916614844 L9.67284723,8.01491201 L9.15588441,0.916614844 C9.15588441,0.916614844 7.37499587,-0.81644658 2.12744709,0.48294645 L2.12744709,12.0829042 C2.12744709,12.0829042 6.89296297,11.6014087 8.33744951,12.3725539 C8.33744951,12.3725539 8.29284669,12.9427177 9.66048741,12.950241 L9.67284723,12.950241 L9.68574443,12.950241 C11.0528478,12.9427177 11.0082449,12.3725539 11.0082449,12.3725539 C12.4527315,11.6014087 17.2182474,12.0829042 17.2182474,12.0829042 L17.2182474,0.48294645 C11.9706986,-0.81644658 10.1898101,0.916614844 10.1898101,0.916614844\\" transform=\\"translate(2.327 4.245)\\"></path><path d=\\"M0.91156344,2.55923464 L0.885231654,13.9028606 L0.455862327,13.4729539 L7.29245378,13.4192155 C7.34887904,13.421365 7.39186971,13.4697296 7.38972017,13.52508 C7.38649587,13.5777436 7.3435052,13.6196595 7.29245378,13.6218091 L0.455862327,14.3333047 C0.229623921,14.3440523 0.0372406687,14.1694027 0.0270303844,13.9442391 L0.0259556176,13.925968 L0.0259556176,13.9028606 L0.000161215016,2.55923464 C-0.000376168371,2.3082766 0.203292135,2.10353353 0.453712794,2.10245876 C0.705745602,2.101384 0.911026056,2.30612707 0.91156344,2.55708511 L0.91156344,2.55923464 Z\\" transform=\\"translate(2.327 4.245)\\"></path><path d=\\"M19.3458019,2.55923464 L19.3200075,13.9028606 L19.3200075,13.9248933 C19.3183954,14.1516691 19.1351477,14.3343794 18.9089092,14.3343794 L18.8895634,14.3333047 L12.0535094,13.6218091 C11.9970841,13.6196595 11.9540934,13.571295 11.956243,13.5159445 C11.9594673,13.463281 12.002458,13.421365 12.0535094,13.4192155 L18.8895634,13.4729539 L18.4607315,13.9028606 L18.4343997,2.55923464 C18.4333249,2.3082766 18.6375306,2.10353353 18.8879513,2.10245876 C19.1405215,2.101384 19.3452646,2.30612707 19.3458019,2.55708511 L19.3458019,2.55923464 Z\\" transform=\\"translate(2.327 4.245)\\"></path></g></svg></div></div>
        <div class=\\"hig__global-nav__side-nav__section__group__module__link__title\\">title</div>
    </a>
    <div class=\\"hig__global-nav__side-nav__section__group__module__collapse hig__global-nav__side-nav__section__group__module__collapse--collapsed\\"><div class=\\"hig__icon\\"><svg width=\\"10\\" height=\\"6\\" viewBox=\\"0 0 10 6\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"caret\\"><title>hig/icon/caret</title><polyline points=\\"8 11 12 15 16 11\\" transform=\\"translate(-6.815 -10)\\" stroke-width=\\"1.5\\" stroke=\\"#979797\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></polyline></svg></div></div><!--MODULE-COLLAPSE-->
    <div class=\\"hig__global-nav__side-nav__section__group__module__submodules\\">
        <div class=\\"hig__global-nav__side-nav__section__group__module__submodule hig__global-nav__side-nav__section__group__module__submodule--hide\\">
    <a class=\\"hig__global-nav__side-nav__section__group__module__submodule__link\\" href=\\"#\\">Document New</a>
</div><!--SUBMODULE-->
    </div>
</div><!--MODULE-->
</div><!--GROUP-->
</div><!--SECTION-->
        <div class=\\"hig__global-nav__sidenav__links\\">
            <!--LINK-->
        </div>
        <div class=\\"hig__global-nav__sidenav__copyright\\">Â© 2017 Autodesk, Inc.</div>
    </div>
    <div class=\\"hig__global-nav__side-nav__search\\">
    <div class=\\"hig__global-nav__side-nav__search__icon\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"search-small\\"><title>hig/icon/search-small</title><path d=\\"M15.8702459,15.1631391 L18.9748737,18.267767 C19.1701359,18.4630291 19.1701359,18.7796116 18.9748737,18.9748737 C18.7796116,19.1701359 18.4630291,19.1701359 18.267767,18.9748737 L15.2403215,15.9474283 C14.0545271,17.2108151 12.3694256,18 10.5,18 C6.91014913,18 4,15.0898509 4,11.5 C4,7.91014913 6.91014913,5 10.5,5 C14.0898509,5 17,7.91014913 17,11.5 C17,12.8587306 16.5831026,14.1200897 15.8702459,15.1631391 Z M16,11.5 C16,8.46243388 13.5375661,6 10.5,6 C7.46243388,6 5,8.46243388 5,11.5 C5,14.5375661 7.46243388,17 10.5,17 C13.5375661,17 16,14.5375661 16,11.5 Z\\" fill=\\"#353636\\" fill-rule=\\"evenodd\\"></path></svg></div></div>
    <div class=\\"hig__global-nav__side-nav__search__inputholder\\"><input class=\\"hig__global-nav__side-nav__search__inputholder__input\\" type=\\"text\\" placeholder=\\"Search for something\\" value=\\"\\"></div>
    <div class=\\"hig__global-nav__side-nav__search__clear\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-small\\"><title>hig/icon/close-small</title><g fill=\\"none\\" fill-rule=\\"evenodd\\"><circle stroke=\\"#353636\\" cx=\\"12\\" cy=\\"12\\" r=\\"8\\"></circle><polygon fill=\\"#353636\\" transform=\\"rotate(45 12 12)\\" points=\\"16.5 11.4518414 12.5847458 11.4518414 12.5847458 7.5 11.4152542 7.5 11.4152542 11.4518414 7.5 11.4518414 7.5 12.5226629 11.4152542 12.5226629 11.4152542 16.5 12.5847458 16.5 12.5847458 12.5226629 16.5 12.5226629\\"></polygon></g></svg></div></div>
</div><!--SEARCH-->
</div><!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;
